### PR TITLE
Fix screenshot taking on platforms with hardware bitmaps

### PR DESCRIPTION
**Pull request by lfkellogg**
Before: http://go/scrcast/NTM2MTE4NTkwMjgyMTM3NnxlMzFkZTdmNy0yYg
After: http://go/scrcast/NjE2MzAwNzY3NjYxMjYwOHwwNDg0N2I2Ny0wYw

### CODE REVIEW

File: firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/ScreenshotTaker.java [hunk 1]
```diff
@@ -93,19 +99,58 @@ Task<Bitmap> captureScreenshot() {
...
+    Canvas canvas = new Canvas(bitmap);
+    activity.getWindow().getDecorView().getRootView().draw(canvas);
+    return Tasks.forResult(bitmap);
+  }
+
+  @TargetApi(Build.VERSION_CODES.O)
+  private static Task<Bitmap> captureScreenshotOreo(Activity activity) {
+    Bitmap bitmap = getBitmapForScreenshot(activity);
+    TaskCompletionSource<Bitmap> taskCompletionSource = new TaskCompletionSource<>();
+    LogWrapper.getInstance().e("LKELLOGG: NEW PATH HERE");
```
Comment (kaibolay) [inline firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/ScreenshotTaker.java:130]: 😃

File: firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/ScreenshotTaker.java [hunk 2]
```diff
@@ -93,19 +99,58 @@ Task<Bitmap> captureScreenshot() {
...
+    try {
+      // PixelCopy can handle Bitmaps with Bitmap.Config.HARDWARE and is the recommended way of
+      // capturing screens moving forward
+      PixelCopy.request(
+          activity.getWindow(),
+          bitmap,
+          (result) -> {
```
Comment (kaibolay) [inline firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/ScreenshotTaker.java:137]: nit: no need for (parentheses)

File: firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/ScreenshotTaker.java [hunk 3]
```diff
@@ -93,19 +99,58 @@ Task<Bitmap> captureScreenshot() {
...
+  }
+
+  @TargetApi(Build.VERSION_CODES.O)
+  private static Task<Bitmap> captureScreenshotOreo(Activity activity) {
+    Bitmap bitmap = getBitmapForScreenshot(activity);
+    TaskCompletionSource<Bitmap> taskCompletionSource = new TaskCompletionSource<>();
+    LogWrapper.getInstance().e("LKELLOGG: NEW PATH HERE");
+    try {
+      // PixelCopy can handle Bitmaps with Bitmap.Config.HARDWARE and is the recommended way of
+      // capturing screens moving forward
```
Comment (kaibolay) [inline firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/ScreenshotTaker.java:133]: nit: Not sure if we need this (entire) comment. Maybe just abbreviate to: "PixelCopy can handle Bitmap.Config.HARDWARE".

**Review by kaibolay — APPROVED** [review]

File: firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/ScreenshotTaker.java (hunk 1 above)
Comment (lfkellogg) [reply firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/ScreenshotTaker.java:130]: Oops!!!

**Pull request merged**